runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -576,9 +576,32 @@ bool J9::ValuePropagation::transformUnsafeCopyMemoryCall(TR::Node *arraycopyNode
       TR::TreeTop *tt = _curTree;
       TR::Node *ttNode = tt->getNode();
 
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+      if (TR::Compiler->om.isOffHeapAllocationEnabled()
+            && (ttNode->getOpCodeValue() == TR::treetop || ttNode->getOpCode().isResolveOrNullCheck()))
+         {
+         OMR::ValuePropagation *vp = this;
+         ListIterator<OMR::ValuePropagation::TR_TreeTopNodePair> copyMemoryIt(&vp->_offHeapCopyMemory);
+         bool alreadyAdded = false;
+         OMR::ValuePropagation::TR_TreeTopNodePair *copyMemoryPair;
+         for (copyMemoryPair = copyMemoryIt.getFirst();
+            copyMemoryPair; copyMemoryPair = copyMemoryIt.getNext())
+            {
+            if (copyMemoryPair->_node == arraycopyNode)
+               {
+               alreadyAdded = true;
+               break;
+               }
+            }
+
+         if (!alreadyAdded)
+            vp->_offHeapCopyMemory.add(new (vp->comp()->trStackMemory()) OMR::ValuePropagation::TR_TreeTopNodePair(tt, arraycopyNode));
+         return false;
```

Comment:
I think this can return true. The transformation hasn't been carried out per se, but the node has been recognized as an unsafe memory copy call, the transformation has been arranged for, and there's nothing else for the VP handler to do with this node

Reply:
Okay 👍